pkg/state/state_cni_plugins.go

```diff
@@ -114,7 +115,8 @@ func (s *stateCNIPlugins) GetWatchSources() map[string]*source.Kind {
 func (s *stateCNIPlugins) getManifestObjects(
 	cr *mellanoxv1alpha1.NicClusterPolicy) ([]*unstructured.Unstructured, error) {
 	renderData := &CNIPluginsManifestRenderData{
-		CrSpec: cr.Spec.SecondaryNetwork.CniPlugins,
+		CrSpec:           cr.Spec.SecondaryNetwork.CniPlugins,
+		NicClusterPolicy: cr.Spec,
 		RuntimeSpec: &runtimeSpec{
```

Comment:
We should not pass the whole NicClusterPolicy object to template. Let's pass only NodeAffinity or move it under RuntimeSpec field